include/rapidcheck/state/Commands.hpp

```diff
@@ -1,9 +1,155 @@
 #pragma once
 
 #include "rapidcheck/detail/Results.h"
+#include "rapidcheck/state/gen/ParallelCommands.hpp"
+#include "rapidcheck/Assertions.h"
+#include <thread>
+#include <mutex>
+#include "rapidcheck/detail/Barrier.h"
 
 namespace rc {
 namespace state {
+namespace detail {
+
+template <typename Model, typename Cmd>
+void allInterleavingsAreValid(const Commands<Cmd> &left,
```

Comment:
This one is a bit weirdly named... it's not really a predicate since the return type is void.